lib/measured/conversion_builder.rb

```diff
@@ -17,28 +17,44 @@ def unit(unit_name, aliases: [], value:)
   end
 
   def case_sensitive(value)
-    @case_sensitive = !!value
+    if @case_sensitive != !!value
+      @case_sensitive = !!value
+
+      # build_unit references @units and @base_unit for doing the duplicate name check, so we need to empty
+      # them before rebuilding the units list to ensure it doesn't raise exceptions unnecessarily
+      old_units = @units
+      old_base_unit = @base_unit
+      @units = []
+      @base_unit = nil
+
+      build_unit(old_base_unit.name, aliases: old_base_unit.aliases, value: old_base_unit.value) if old_base_unit
+      @units = old_units.map { |unit| build_unit(unit.name, aliases: unit.aliases, value: unit.value) }
```

Comment:
I can see two other options we can take here:

1. Don't build anything until the end, just collect all the information and lazily build (we opted against this to more easily fail as soon as possible).
2. Don't have a declarative DSL declarative by raising when case_sensitive is called after a unit has been added.